### PR TITLE
make wcm search user directories for icons

### PR DESCRIPTION
**Pull request by epsilon-0**

### CODE REVIEW

File: src/wcm.cpp [hunk 1]
```diff
@@ -1697,3 +1698,33 @@ std::shared_ptr<wf::config::section_t> WCM::get_config_section(Plugin *plugin)
...
+    std::string icon_path;
+    std::string xdg_data_dir;
+    char *c_xdg_data_dir = std::getenv("XDG_DATA_HOME");
+    char *c_user_home    = std::getenv("HOME");
+
+    if (c_xdg_data_dir != NULL)
+    {
+        xdg_data_dir = c_xdg_data_dir;
+    } else
+    {
```
Comment (NamorNiradnug) [inline src/wcm.cpp:1714]: Extra braces here, please use `if else`.

File: src/wcm.cpp [hunk 2]
```diff
@@ -1697,3 +1698,33 @@ std::shared_ptr<wf::config::section_t> WCM::get_config_section(Plugin *plugin)
...
+        if (c_user_home != NULL)
+        {
+            xdg_data_dir = (std::string)c_user_home + "/.local/share/";
+        }
+    }
+
+    if (xdg_data_dir.size())
```
Comment (NamorNiradnug) [inline src/wcm.cpp:1721]: Please use `std::string::empty` method instead of comparing `size` with zero.

**Review by NamorNiradnug — CHANGES_REQUESTED** [review]
Hi! Aside from requested changes, LGTM, thanks on your work on WCM!

File: src/wcm.cpp (hunk 1 above)
Comment (epsilon-0) [reply src/wcm.cpp:1714]: done

File: src/wcm.cpp (hunk 2 above)
Comment (epsilon-0) [reply src/wcm.cpp]: done

**Review by NamorNiradnug — APPROVED** [review]
Thanks!

**Review by NamorNiradnug — APPROVED** [review]

**Comment by soreau** [comment]
Thanks!

**Pull request merged**